### PR TITLE
upgrade tiflash after pd

### PR DESCRIPTION
**Pull request by DanielZhangQD**
### What problem does this PR solve?

Fix https://github.com/pingcap/tidb-operator/issues/4096
### What is changed and how does it work?

Sync TiFlash after pd
### Code changes

- [x] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [x] Manual test 
  - Deploy TidbCluster with CDC, PD, TiKV, TiFlash, Pump, TiDB, waiting for Pods ready, upgrade the tc, check that the upgrade order is CDC, PD, TiFlash, TiKV, Pump, TiDB
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [x] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note

```

### CODE REVIEW

File: pkg/manager/member/pd_upgrader.go [hunk 1]
```diff
@@ -45,7 +45,6 @@ func (u *pdUpgrader) gracefulUpgrade(tc *v1alpha1.TidbCluster, oldSet *apps.Stat
 		return fmt.Errorf("tidbcluster: [%s/%s]'s pd status sync failed, can not to be upgraded", ns, tcName)
 	}
 	if tc.Status.TiCDC.Phase == v1alpha1.UpgradePhase ||
-		tc.Status.TiFlash.Phase == v1alpha1.UpgradePhase ||
```
Comment (csuzhangxc) [inline pkg/manager/member/pd_upgrader.go:48]: update the log with `tiflash status` too?

Comment (DanielZhangQD) [reply pkg/manager/member/pd_upgrader.go:48]: @csuzhangxc Fixed, PTAL again.

**Review by KanShiori — APPROVED** [review]
LGTM

**Review by csuzhangxc — APPROVED** [review]

**Pull request merged**